src/lib/expression/evaluation/like_matcher.cpp

```diff
@@ -86,20 +86,19 @@ LikeMatcher::AllPatternVariant LikeMatcher::pattern_string_to_pattern_variant(co
     return ContainsPattern{std::get<pmr_string>(tokens[1])};
 
   } else {
-    /**
-     * Pattern is either MultipleContainsPattern, e.g., '%hello%world%how%are%you%' or we fall back to
-     * using a regex matcher.
-     *
-     * A MultipleContainsPattern begins and ends with '%' and  contains only strings and '%'.
-     */
-
-    // Pick ContainsMultiple or Regex
-    auto pattern_is_contains_multiple = true;  // Set to false if tokens don't match %(, string, %)* pattern
-    auto strings = std::vector<pmr_string>{};  // arguments used for ContainsMultiple, if it gets used
-    auto expect_any_chars = true;              // If true, expect '%', if false, expect a string
+    // Pattern is either MultipleContainsPattern, e.g., 'hello%world%' or '%hello%world%how%are%you%'. If not, we fall
+    // back to using a regex matcher.
+    // A MultipleContainsPattern can begin with '%', ends with '%', and contains only strings and '%'.
+
+    auto pattern_is_contains_multiple = true;  // Set to false if tokens don't match %(, string, %)* pattern.
+    auto strings = std::vector<pmr_string>{};  // Arguments used for ContainsMultiple, if it gets used.
+    // starts_with_any_char is used to denote whether the patterns starts with `%` or not.
+    const auto starts_with_any_char = tokens.empty() ? false : (tokens[0] == PatternToken{Wildcard::AnyChars});
+    // If true, expect '%', if false, expect a string. Used to check that we alternate between string tokens and '%'.
+    auto expect_any_chars = starts_with_any_char;
 
     // Check if the tokens match the layout expected for MultipleContainsPattern - or break and set
-    // pattern_is_contains_multiple to false once they don't
+    // pattern_is_contains_multiple to false once they don't.
     for (const auto& token : tokens) {
```

Comment:
Maybe we can simplify this loop a little bit. Here is some pseudo-code:

```cpp
for (auto i = size_t{0}; i < tokens.length() - 1; i++) {
  if (std::holds_alternative<pmr_string>(tokens[i])) {
    if (tokens[i + 1] != PatternToken{Wildcard::AnyChars}){
      pattern_is_contains_multiple = false;
      break;
    }
    strings.emplace_back(std::get<pmr_string>(tokens[i]));
    }
  if (tokens[i] == PatternToken{Wildcard::AnyChars}) {
    if (std::holds_alternative<pmr_string>(tokens[i + 1])){
      pattern_is_contains_multiple = false;
      break;
    }
  }
}
```

The main idea is that we just check that the next token is allowed. In the current version, I find it a little bit hard to read with the logical expressions and `expect_any_chars`.